pkg/repro/repro.go

```diff
@@ -603,126 +605,38 @@ func (ctx *context) returnInstance(inst *reproInstance) {
 }
 
 func (ctx *context) reproLogf(level int, format string, args ...interface{}) {
+	if ctx.logf != nil {
+		ctx.logf(format, args...)
+	}
 	prefix := fmt.Sprintf("reproducing crash '%v': ", ctx.crashTitle)
 	log.Logf(level, prefix+format, args...)
 	ctx.stats.Log = append(ctx.stats.Log, []byte(fmt.Sprintf(format, args...)+"\n")...)
 }
 
 func (ctx *context) bisectProgs(progs []*prog.LogEntry, pred func([]*prog.LogEntry) (bool, error)) (
 	[]*prog.LogEntry, error) {
+	// Set up progs bisection.
 	ctx.reproLogf(3, "bisect: bisecting %d programs", len(progs))
-
-	ctx.reproLogf(3, "bisect: executing all %d programs", len(progs))
-	crashed, err := pred(progs)
-	if err != nil {
-		return nil, err
-	}
-	if !crashed {
-		ctx.reproLogf(3, "bisect: didn't crash")
-		return nil, nil
-	}
-
-	guilty := [][]*prog.LogEntry{progs}
-again:
-	if len(guilty) > 8 {
-		// This is usually the case for flaky crashes. Continuing bisection at this
-		// point would just take a lot of time and likely produce no result.
+	ret, err := generic.MinimizeSlice(generic.MinimizeConfig[*prog.LogEntry]{
+		Pred: func(progs []*prog.LogEntry) (bool, error) {
```

Comment:
Please move the predicate out of line. "Configs" are supposed to be declarative and readable, but it's hard to read with significnat code chunks.